src/cascadia/TerminalApp/SuggestionsControl.cpp

```diff
@@ -1020,14 +1101,67 @@ namespace winrt::TerminalApp::implementation
     void SuggestionsControl::_setDirection(TerminalApp::SuggestionsDirection direction)
     {
         _direction = direction;
+
+        // We need to move either the list of suggestions, or the tooltip, to
+        // the top of the stack panel (depending on the layout).
+        Grid controlToMoveToTop = nullptr;
+
         if (_direction == TerminalApp::SuggestionsDirection::TopDown)
         {
             Controls::Grid::SetRow(_searchBox(), 0);
+            controlToMoveToTop = _backdrop();
         }
         else // BottomUp
         {
             Controls::Grid::SetRow(_searchBox(), 4);
+            controlToMoveToTop = _descriptionsBackdrop();
         }
+
+        assert(controlToMoveToTop);
+        const auto& children{ _listAndDescriptionStack().Children() };
+        uint32_t index;
+        if (children.IndexOf(controlToMoveToTop, index))
+        {
+            children.Move(index, 0);
+        }
+    }
+
+    void SuggestionsControl::_recalculateTopMargin()
+    {
+        auto currentMargin = Margin();
+        // Call Measure() on the descriptions backdrop, so that it gets it's new
+        // DesiredSize for this new description text.
+        //
+        // If you forget this, then we _probably_ weren't laid out since
+        // updating that text, and the ActualHeight will be the _last_
+        // description's height.
+        const til::size actualSize{ til::math::rounding, ActualWidth(), ActualHeight() };
+        _descriptionsBackdrop().Measure(actualSize.to_winrt_size());
```

Comment:
We shouldn't cast to `til::size` as an intermediary here. That we do this elsewhere is one of the reasons IMO why we have these tiny pixel shifts and inaccuracies at >100% display scale.

This should IMO be:
```cpp
_descriptionsBackdrop().Measure({
    static_cast<float>(ActualWidth()),
    static_cast<float>(ActualHeight()),
});
```

Apropos, isn't it funny how `winrt::Windows::Foundation::Size` uses `float` while `ActualWidth` and a ton of other functions use `double`? Quite the consistent API...

Reply:
And how `ActualWidth` and `ActualHeight` are separate members like this, rather than an `ActualSize`? 🤬